src/test/java/com/google/cloud/tools/managedcloudsdk/command/AsyncByteConsumerTest.java

```diff
@@ -82,7 +82,9 @@ public void testConsumeBytes() throws Exception {
     int count = bytes.getAllValues().size();
     StringBuilder result = new StringBuilder();
     for (int i = 0; i < count; i++) {
-      result.append(new String(bytes.getAllValues().get(i), 0, nBytes.getAllValues().get(i)));
+      byte[] rawBytes = bytes.getAllValues().get(i);
+      int length = nBytes.getAllValues().get(i);
+      result.append(new String(rawBytes, 0, length, StandardCharsets.UTF_8));
```

Comment:
Although this is likely to always work in practice, I don't see any guarantee that the consumer will receive bytes that align on a character boundary.  In which case, this `new String()` could fail.  Both `AsyncStreamSaverFactory.CollectingByteHandler` and `AsyncStreamHandlerFactory.ConsoleListenerForwardingByteHandler` suffer from this in particular.

Reply:
At least we're OK in tests.